src/H5FDs3comms.c

```diff
@@ -776,7 +802,7 @@ H5FD__s3comms_s3r_open(const char *url, const H5FD_ros3_fapl_t *fa, const char *
         HGOTO_ERROR(H5E_VFL, H5E_BADVALUE, NULL, "error while setting CURL option (CURLOPT_HTTPGET)");
     if (CURLE_OK != curl_easy_setopt(curlh, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1))
         HGOTO_ERROR(H5E_VFL, H5E_BADVALUE, NULL, "error while setting CURL option (CURLOPT_HTTP_VERSION)");
-    if (CURLE_OK != curl_easy_setopt(curlh, CURLOPT_FAILONERROR, 1L))
+    if (CURLE_OK != curl_easy_setopt(curlh, CURLOPT_FAILONERROR, 0L)) /* Defaults off anyway, but make sure */
```

Comment:
This option makes error handling slightly simpler, but also throws away the HTTP response body that might contain more details on what a client should do for a particular error. To handle these situations in the future, it's better to make sure this option is turned off and then directly handle HTTP status codes ourself.